pkg/service/backup_scheduler.go

```diff
@@ -138,26 +138,31 @@ func (h *BackupHandler) runFullBackup(now time.Time) {
 		backupSkippedCounter.Inc()
 		return
 	}
-	if !h.fullBackupInProgress.CompareAndSwap(false, true) {
-		slog.Debug("Backup is currently in progress, skipping full backup", "name", h.backupRoutine.Name)
+	if !h.isFullEligible(now, h.state.LastRun) {
+		slog.Debug("The full backup is not due to run yet", "name", h.backupRoutine.Name)
 		return
 	}
-	// release the lock
-	defer h.fullBackupInProgress.Store(false)
 
-	if !h.isFullEligible(now, h.state.LastRun) {
-		slog.Debug("The full backup is not due to run yet", "name", h.backupRoutine.Name)
+	if !h.fullBackupInProgress.CompareAndSwap(false, true) {
+		slog.Debug("Full backup is currently in progress, skipping full backup", "name", h.backupRoutine.Name)
 		return
 	}
+	slog.Debug("Acquire fullBackupInProgress lock", "name", h.backupRoutine.Name)
```

Comment:
This debug line might help understand when backup is happening; We might want to delete it closer to release